### ISSUE TITLE
Setting contentEditable without setAttribute does not set the attribute in js-dom

### ISSUE THREAD

**Issue by carlesnunez** [kitten/use-editable#13]
Hello, I've been using your hook on a project feature and when writting tests I noticed that the attribute was not being set. Performing a bit of investigation I detected that we set the contentEditable attribute by doing the next:

```
element.contentEditable = 'BLA BLA'
```
After testing it I noticed that even though my test is getting inside the function the property was not being set. So I decided to modify the library and test if using setAttribute fixed the issue and it worked:

```
element.setAttribute('contenteditable', 'plaintext-only');
```

What's the reason behind setting directly the property?

**Comment by kitten** [comment]
I think you may have to add someore details of what you're attempting. It's generally not intended for you to manage the `contentEditable` state of your library manually, since it's managed by the hook: https://github.com/kitten/use-editable/blob/852dc60c37da71e11519bd426546ab317f55a1f9/src/useEditable.ts#L296

**Comment by carlesnunez** [comment]
Hello @kitten and thanks for the fast reply!

I was just ensuring that the component had the attribute via react-testing-library but the attribute is never set :S

**Comment by kitten** [comment]
This may be a limitation in JSDOM, I'm not sure. But it's something that's either way not strictly speaking testable, because I'd doubt that JSDOM can emulate content editables and focus behaviour completely. In fact, I'm actually sure that JSDOM isn't capable of fully emulating the browser's behaviour on focus behaviour even, so it goes to reason it won't support content editables either 😅 I'd advise against testing `useEditable` on your end in general, as it'd be outside of your codebase's responsibility.

In this case, it's also not possible to switch to `setAttribute('contenteditable', ...)` as it's not the same behaviour. In the snippet I've pasted it's used to differentiate between browsers that support plaintext-only mode and those that don't. Setting `contentEditable` directly will throw an error for unsupported values, which makes this detection possible. By definition, `setAttribute` won't throw these kinds of errors since it acts on DOM attributes directly.

There's some things to be said about `use-editable` maybe needing more tests in the future, which has been difficult in the past due to a lack of tools that would do the job, but I've actually now worked on another library that does some complex focus management, where I've used Cypress' component testing feature together with [`cypress-real-events`](https://github.com/dmtrKovalenko/cypress-real-events), so that could take over the unit testing capabilities at least for this library.

But yea, to hint at the earlier point, if you're testing this in your code base, I'd generally advise against testing an external library beyond trivial checks. But Cypress in general (or something similar, e.g. raw puppeteer tests) would be needed to emulate the full browser behaviour here.
